compiler/rustc_metadata/src/rmeta/mod.rs

```diff
@@ -419,10 +419,10 @@ define_tables! {
     lookup_deprecation_entry: Table<DefIndex, LazyValue<attr::Deprecation>>,
     explicit_predicates_of: Table<DefIndex, LazyValue<ty::GenericPredicates<'static>>>,
     generics_of: Table<DefIndex, LazyValue<ty::Generics>>,
-    explicit_super_predicates_of: Table<DefIndex, LazyValue<ty::GenericPredicates<'static>>>,
+    explicit_super_predicates_of: Table<DefIndex, LazyArray<(ty::Clause<'static>, Span)>>,
```

Comment:
```suggestion
    explicit_super_predicates_of: Table<DefIndex, LazyArray<ty::EarlyBinder<'static, (ty::Clause<'static>, Span)>>>,
```

I think I prefer encoding it as such (same for the one below). There's precedent and it would allow us to the revert the explicit `.skip_binder()`s in `compiler/rustc_metadata/src/rmeta/encoder.rs`.

Reply:
I don't believe so? But I guess I'll check.

Reply:
No, it does not